backend/src/common/plugins/auth-plugin.ts

```diff
@@ -0,0 +1,47 @@
+import fastifyPlugin from 'fastify-plugin';
+import { UserValidationMessage } from 'shared';
+
+import { PluginName } from '~/common/enums/enums.js';
+import {
+    extractApiPath,
+    extractTokenFromHeaders,
+} from '~/common/helpers/helpers.js';
+import { HttpCode, HttpError } from '~/common/http/http.js';
+
+import { type AuthPluginOptions } from './types/types.js';
+
+const authPlugin = fastifyPlugin(
+    (fastify, { jwtService, whitelistedRoutes }: AuthPluginOptions, done) => {
+        fastify.decorateRequest('user', null);
+        const whiteList = whitelistedRoutes.map((route) =>
+            extractApiPath(route),
+        ) as string[];
+
+        fastify.addHook('preHandler', async (request) => {
+            const extractedApiPath = extractApiPath(request.routeOptions.url);
+            if (extractedApiPath && whiteList?.includes(extractedApiPath)) {
+                return;
+            }
```

Comment:
Seems like you don't need this extractApiHelper at all,
You could just replace check with
`whitelistedRoutes.includes(request.routeOptions.urs)`
Or this is not going to work?
Also this optional chaining near `whiteList` is present because linter enforces you to use Set? Right? 
Probably should whitelistedRoutes could be Set<string> in this case

Reply:
we don't need optional chaining at all